### DISCUSSION TITLE
How to determine the user's role when logging in

### DISCUSSION THREAD

**Discussion by Zhaibian1** [cerbos/cerbos#1034] [Q&A] [answered]
How to determine the user's role when logging in

**Comment by charithe** [comment, marked as answer]
Hi, determining a user's roles is something that should happen at the authentication stage. Typically people use an identity provider (IdP) like Auth0, Cognito, LDAP or user-management modules provided by their chosen framework such as Django, Spring etc.

Cerbos does not handle authentication. It's an authorization solution that expects your application to perform the authentication step and resolve all the roles that user (principal) has. You then send that information in the Cerbos API request and Cerbos evaluates the access rules for you. 

We have several articles and demos showing how to integrate with various IdPs. Here are a few of those.

- https://cerbos.dev/blog/page/1?tag=integration
- https://github.com/cerbos/express-auth0-cerbos
- https://github.com/cerbos/python-cognito-cerbos
